packages/e2e-tests/helpers/console-panel.ts

```diff
@@ -119,9 +119,11 @@ export function findConsoleMessage(
     ? `[data-test-message-type="${messageType}"]`
     : '[data-test-name="Message"]';
 
-  return expected
+  const messageLocator = expected
     ? page.locator(`${attributeSelector}:has-text("${expected}")`)
     : page.locator(`${attributeSelector}`);
+  // TODO This would be better replaced by an `{exact? boolean}` option threaded through
+  return messageLocator.first();
```

Comment:
I think you should return all matching messages and let the _caller_ decide if they want only one of them (`.findConsoleMessages(...).first()`) or if they want to do something on all matches (like assert an expected _count_ etc)